### PR TITLE
[15.0][FIX] base_report_to_printer: Report send to printer when is rendered from email template

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT51079

ping @chienandalu @pilarvargas-tecnativa

### CODE REVIEW

File: base_report_to_printer/models/mail_template.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    def generate_email(self, res_ids, fields):
+        """Ensure that any report rendered to send it by email not is sent to printer
+        directly if the report is setting to send it to printer.
+        For example, user set delivery slip report redirect to printer and
+        'Send an email when picking is validated' is active, if we do not inject this
+        context the report will be send twice to printer.
+        """
+        return super(
+            MailTemplate, self.with_context(must_skip_send_to_printer=True)
+        ).generate_email(res_ids, fields)
```
Comment (pilarvargas-tecnativa) [inline base_report_to_printer/models/mail_template.py:19]: I believe this solution is already covered in a module that is not migrated.
https://github.com/OCA/report-print-send/blob/14.0/base_report_to_printer_mail/models/mail_template.py

I don't know if it is convenient to keep it separate and migrate the other module.

**Comment by sergio-teruel** [comment]
What?? There is a module to fix other module??
I think that if you install base_report_to_printer in any instance the mail module allways will be installed, this is the reason by adding web dependency and fix it inside the module...

**Comment by pilarvargas-tecnativa** [comment]
> What?? There is a module to fix other module?? I think that if you install base_report_to_printer in any instance the mail module allways will be installed, this is the reason by adding web dependency and fix it inside the module...

Then that module should disappear with a merge script. I leave it as a comment in case someone uses it, not in our case.

**Review by pilarvargas-tecnativa — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
That module exists because the dependency in this module is `web`, not `mail`. You may say that both modules are always installed, but that's not really something that you can assure, so I think that changing the dependency mustn't be done here, but on a new version, and for now, to migrate the other module.

**Review by chienandalu — COMMENTED** [review]
I agree that is quite absurd to have that glue module, but let's migrate this and move on...

**Comment by sergio-teruel** [comment]
Migration of module https://github.com/OCA/report-print-send/pull/374

**Pull request closed without merging**